opensuse-tomcat-juli-image/src/main/docker/Dockerfile

```diff
@@ -0,0 +1,74 @@
+#
+# Copyright 2017-2022 Micro Focus or one of its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Docker registry
+ARG DOCKER_HUB_PUBLIC=docker.io
+
+#
+# Preliminary image that downloads Tomcat and makes some modifications
+#
+FROM ${DOCKER_HUB_PUBLIC}/opensuse/leap:latest AS builder
+RUN zypper -n install gzip tar xsltproc
+WORKDIR /wd
+
+ENV TOMCAT_PARENT_DIR=/wd/root/usr/share \
+    TOMCAT_DIR_NAME=tomcat
+ENV TOMCAT_ROOT_DIR $TOMCAT_PARENT_DIR/$TOMCAT_DIR_NAME
+ENV TOMCAT_CONF_DIR $TOMCAT_ROOT_DIR/conf
+
+# Extract Tomcat and remove unwanted web applications
+COPY apache-tomcat-9.0.64.tar.gz .
+RUN mkdir -p $TOMCAT_PARENT_DIR && \
+    cd $TOMCAT_PARENT_DIR && \
+    tar xzf /wd/apache-tomcat-9.0.64.tar.gz && \
+    mv apache-tomcat-9.0.64 $TOMCAT_DIR_NAME && \
+    cd $TOMCAT_ROOT_DIR/webapps && \
+    rm -rf docs/ examples/ host-manager/ manager/
+
+# Overwrite the main logging configuration file
+# (It is being copied to the working folder first, and then overwritten using cp, in order to retain the original file's permissions.)
+COPY logging.properties .
+RUN cp logging.properties $TOMCAT_CONF_DIR
+
+# Apply the specified transform to the server.xml file
+COPY server.xslt .
+RUN cp $TOMCAT_CONF_DIR/server.xml .
+RUN xsltproc -o $TOMCAT_CONF_DIR/server.xml server.xslt server.xml
+
+# Add in the startup scripts
+COPY startup.d/ root/startup/startup.d/
+RUN chmod +x root/startup/startup.d/*
+
+#
+# The actual image definition
+#
+FROM ${DOCKER_HUB_PUBLIC}/cafapi/opensuse-jre8:3
+
+RUN zypper -n refresh && \
+    zypper -n update && \
+    zypper -n clean --all
+
+ENV CATALINA_HOME /usr/share/tomcat
+ENV PATH $CATALINA_HOME/bin:$PATH
+WORKDIR $CATALINA_HOME
+
+HEALTHCHECK CMD curl --fail http://localhost:8081/healthcheck || exit 1
+
+EXPOSE 8080 8081 8443
+CMD ["catalina.sh", "run"] 
+
+COPY maven/ /lib/
```

Comment:
There are a few issues with this copy command;
It adds back in a second copy to the final image, something that was already flagged and resolved. If you wanted this added build the file structure in the first stage.
It is copying everything from the maven folder to /lib, so in the final image you end up with /lib/lib/<file>
What is the purpose of adding a file there, what will use it?
If you wanted the jar to be used by tomcat it needed copied to 'COPY maven/lib/*.jar /usr/share/tomcat/lib/'
But that's not what is required, the slf jar is not needed as part of the juli image, that was the fix to do here.